common/persistence/class.PhpFileDriver.php

```diff
@@ -78,190 +98,352 @@ class common_persistence_PhpFileDriver implements common_persistence_KvDriver, c
      */
     public function connect($id, array $params)
     {
-        $this->directory = isset($params['dir']) 
+        $this->directory = isset($params['dir'])
             ? $params['dir'].($params['dir'][strlen($params['dir'])-1] === DIRECTORY_SEPARATOR ? '' : DIRECTORY_SEPARATOR)
             : FILES_PATH.'generis'.DIRECTORY_SEPARATOR.$id.DIRECTORY_SEPARATOR;
         $this->levels = isset($params['levels']) ? $params['levels'] : self::DEFAULT_LEVELS;
         $this->humanReadable = isset($params['humanReadable']) ? $params['humanReadable'] : false;
+
+        // Sets ttl mode TRUE when the passed ttl mode is true.
+        $this->setTtlMode(
+            (isset($params[static::OPTION_TTL]) && $params[static::OPTION_TTL] == true)
```

Comment:
Why do we need an optional parameter to enable ttl? Don't we want this solution globally?

Reply:
the data format changes when using TTL, so in order to be backward compatible we need to either implement it as a new solution, or make it a parameter

Reply:
okay, I see.